commit 7e2f9d839f5af447a0fc7fd178dd5d8f58e489bb
Author: Christoph Feck <cfeck@kde.org>
Date:   Wed Dec 7 20:12:59 2016 +0100

    Render text at primary font's baseline
    
    This only addresses the vertical shift and cutoff of lower
    parts of characters, but not horizontal shift (characters
    moving to the left or right).
    
    See the linked review request for details.
    
    This commit is for the 17.04 version. If you can test master
    builds, please add feedback.
    
    REVIEW: 129281
    CCBUG: 371687

--- a/src/TerminalDisplay.cpp
+++ b/src/TerminalDisplay.cpp
@@ -855,18 +855,10 @@
         // This still allows RTL characters to be rendered in the RTL way.
         painter.setLayoutDirection(Qt::LeftToRight);
 
-        // the drawText(rect,flags,string) overload is used here with null flags
-        // instead of drawText(rect,string) because the (rect,string) overload causes
-        // the application's default layout direction to be used instead of
-        // the widget-specific layout direction, which should always be
-        // Qt::LeftToRight for this widget
-        //
-        // This was discussed in: http://lists.kde.org/?t=120552223600002&r=1&w=2
         if (_bidiEnabled) {
-            painter.drawText(rect, 0, text);
+            painter.drawText(rect.x(), rect.y() + _fontAscent + _lineSpacing, text);
         } else {
-            // See bug 280896 for more info
-            painter.drawText(rect, Qt::AlignBottom, LTR_OVERRIDE_CHAR + text);
+            painter.drawText(rect.x(), rect.y() + _fontAscent + _lineSpacing, LTR_OVERRIDE_CHAR + text);
         }
     }
 }
